packages/display/src/DisplayObject.js

```diff
@@ -146,17 +145,38 @@ export default class DisplayObject extends EventEmitter
     }
 
     /**
-     * Updates the object transform for rendering
-     *
-     * TODO - Optimization pass!
+     * Updates the object transform (coordinates and alpha) for rendering
+     * parent must exist
      */
     updateTransform()
     {
-        this.transform.updateTransform(this.parent.transform);
-        // multiply the alphas..
-        this.worldAlpha = this.alpha * this.parent.worldAlpha;
+        const parent = this.parent;
 
-        this._bounds.updateID++;
+        this._boundsID++;
+        // coords transform
+        this.transform.updateTransform(parent.transform);
+        // color/alpha transform
+        this.worldAlpha = this.alpha * parent.worldAlpha;
+    }
+
+    /**
+     * Updates the object transform (coordinates and alpha) for rendering
+     * even if parent is null
+     */
+    safeUpdateTransform()
```

Comment:
Couldn't this all be simplified to and remove the if/else:
```js
const parent = this.parent;
this.parent = parent || this._tempDisplayObjectParent;
this.updateTransform();
this.parent = parent;
```

Reply:
I think its making code less verbose.

Reply:
@bigtimebuddy you can't modify `const` value :)

I would give +1 for `if-else` statement, it improves readability at least twice